src/cim/components/modals/AddBmcHostYamlForm.tsx

```diff
@@ -0,0 +1,208 @@
+        reader.onload = async (readerEvent) => {
+          const content = readerEvent.target?.result;
+          // parse yaml
+          if (typeof content === 'string') {
+            await new Promise((r) => setTimeout(r, 100));
+            let yamlContent;
+            try {
+              yamlContent = jsYaml.loadAll(content);
+              setTextOpenFile(t('ai:Open file'));
+              setShowOpenFileButton(false);
+              setFileName(file.name);
+              setFileError(undefined);
+              setYamlContent(yamlContent);
+            } catch (err) {
+              setTextOpenFile(t('ai:Open file'));
+              setShowOpenFileButton(false);
+              setFileName(file.name);
+              setFileError(
+                'File is not structured correctly. Use the template to use the right file structure.',
+              );
+              setYamlContent([]);
+            }
+          }
+        };
+      }
+    };
+    input.click();
+  };
+
+  const deleteFileContent = () => {
+    setFileError(undefined);
+    setFileName(undefined);
+    setShowOpenFileButton(true);
+    setTextOpenFile(t('ai:Open file'));
+  };
+
+  const handleSubmit: FormikConfig<AddYamlValues>['onSubmit'] = async () => {
+    try {
+      setError(undefined);
+      if (yamlContent !== undefined) {
+        await onCreateBmcByYaml(yamlContent);
+      }
+
+      onClose();
+    } catch (e) {
+      setError(getErrorMessage(e));
+    }
+  };
+
+  return (
+    <Formik initialValues={emptyValues(t)} isInitialValid={false} onSubmit={handleSubmit}>
+      {({ submitForm }: FormikProps<AddYamlValues>) => (
+        <ModalBoxBody>
+          <EmptyState>
+            <Title headingLevel="h4" size="lg">
+              <div>
+                <p>
+                  <UploadIcon size="xl" color="grey" />
+                </p>
+                {t('ai:Upload a YAML file')}
+              </div>
+            </Title>
+            <EmptyStateBody>
+              <div>
+                <p>
+                  {t(
+                    "ai:Upload a YAML file to add hosts's credentials. For each host, include: hostname, Baseboard Mangement Controller Address, username and password",
+                  )}
+                </p>
+
+                <Text
+                  component={TextVariants.a}
+                  isVisitedLink
+                  onClick={() => downloadYaml()}
+                  target="_blank"
+                  className="download_link"
+                >
+                  {t('ai:Download template')}
+                </Text>
+                {showOpenFileButton && (
+                  <Button
+                    variant={ButtonVariant.secondary}
+                    onClick={async () => {
+                      try {
+                        await importBMAsYAML();
+                      } catch (e) {
+                        setError(getErrorMessage(e));
+                      }
+                    }}
```

Comment:
there is eslint warning which you can get rid of by doing
```suggestion
                    onClick={() => {
                          const importYAML = async () => {
                            try {
                              await importBMAsYAML();
                            } catch (e) {
                              setError(getErrorMessage(e));
                            }
                          };
                          void importYAML();
                        }}
```

Reply:
@rawagner I've changed it but eslint error it still appears